src/radical/entk/appman/wfprocessor.py

```diff
@@ -216,28 +224,12 @@ def reset_workflow(self):
     #
     def _execute_workload(self, workload, scheduled_stages):
 
-        # Tasks of the workload need to be converted into a dict
-        # as pika can send and receive only json/dict data
-        wl_json = {'type': 'workload',
-                   'body': [task.as_dict() for task in workload]}
-
-        wl_json = json.dumps(wl_json)
-
-        # Acquire a connection+channel to the rmq server
-        mq_connection = pika.BlockingConnection(self._rmq_conn_params)
-        mq_channel = mq_connection.channel()
+        msg = {'type': 'workload',
+               'body': [task.as_dict() for task in workload]}
```

Comment:
general point about structure of zmq messages - some messages are of format `{'type':.., 'body':..}`, some are `{'type':.., 'object':..}`, and some are just of Task in a dictionary format - would propose to have one structure

Reply:
@mtitov: I started to do that, but realized that it does not make as much sense as I thought it would.  In RP, we have uniform message structured on the pubsub channels: any component can listen on those channels and then needs to figure out what message is 'interesting' and what message can be ignored.  We use the `cmd` field for that which every pubsub message has.  On the queues, we only expect one specific component to listen and that component expects a specific message type on that queue, so we don't have the same structure there.
Same here in EnTK: we use queues, and while the message types differ (task dicts vs. generic objects vs. resource requests), the recipient already knows what message type(s) to expect, and adding more structure adds just noise and code really.
In an abstract way I still agree somewhat: having better defined message structures would make the code easier to inspect for new developers for example, or would make logging on the ZMQ queues simpler as they could better inspect messages, but if we want to do that, then better in sync with RP.
Summary: out of scope for this PR :-)